contracts/ZapV2.sol

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.6.12;
+
+/*
+ * Trader Joe
+ * MIT License; modified from PancakeBunny
+ *
+ */
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./traderjoe/interfaces/IJoePair.sol";
+import "./traderjoe/interfaces/IJoeRouter02.sol";
+import "./traderjoe/interfaces/IWAVAX.sol";
+
+contract ZapV2 is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    /* ========== CONSTANT VARIABLES ========== */
+
+    address public constant WAVAX = 0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7;
+
+    IJoeRouter02 private ROUTER;
+
+    /* ========== INITIALIZER ========== */
+
+    constructor(address _router) public {
+        require(owner() != address(0), "ZapV2: owner must be set");
+
+        ROUTER = IJoeRouter02(_router);
+    }
+
+    receive() external payable {}
+
+    /* ========== External Functions ========== */
+
+    function zapInToken(
+        address tokenFrom,
+        uint256 amountFrom,
+        uint256 amount0Min,
+        uint256 amount1Min,
+        address[] calldata pathToToken0,
+        address[] calldata pathToToken1
+    ) external {
+        IERC20 token = IERC20(tokenFrom);
+        uint256 balanceBefore = token.balanceOf(address(this));
+        IERC20(tokenFrom).safeTransferFrom(msg.sender, address(this), amountFrom);
+        uint256 amount = token.balanceOf(address(this)) - balanceBefore;
+
+        _zapInToken(tokenFrom, amount, amount0Min, amount1Min, pathToToken0, pathToToken1, msg.sender);
+    }
+
+    function zapIn(
```

Comment:
nit but maybe call this `zapInAVAX` to be explicit? similar to how the Router swap methods have `AVAX` in their names, e.g. `swapAVAXForExactTokens`